src/k8s/pkg/k8sd/api/cluster_delete.go

```diff
@@ -0,0 +1,40 @@
+package api
+
+import (
+	"errors"
+	"net/http"
+	"os"
+
+	snaputil "github.com/canonical/k8s/pkg/snap/util"
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/microcluster/state"
+)
+
+func (e *Endpoints) deleteCluster(s *state.State, r *http.Request) response.Response {
```

Comment:
You should handle most of the above logic (except of the microcluster reset).

Reply:
@bschimke95 What do you mean here?